src/Commands/ListCommand.php

```diff
@@ -50,16 +53,53 @@ class ListCommand extends SecretBaseCommand implements SiteAwareInterface
      */
     public function listSecrets($site_id, array $options = ['debug' => false,])
     {
+        $env_name = '';
+        if (strpos($site_id, '.') !== false) {
+            list($site_id, $env_name) = explode('.', $site_id, 2);
+        }
         $site = $this->getSite($site_id);
-        $this->warnIfEnvironmentPresent($site_id);
         $this->setupRequest();
         $secrets = $this->secretsApi->listSecrets($site->id, $options['debug']);
         $print_options = [
             'message' => 'You have no Secrets.'
         ];
+        // If the user requested secrets from a specific environment, then
+        // filter down to just the secret values there.
+        if (!empty($env_name)) {
+            $secrets = $this->secretsForEnv($secrets, $env_name);
+            $print_options = [
+                'message' => "There are no environment overrides in the environment '$env_name'.",
```

Comment:
I thought it might be confusing that way; as implemented, you can easily see if there are overrides. I agree that we could change it as you suggested if desired as follow-on work.